tardis/visualization/tools/sdec_plot.py

```diff
@@ -926,9 +1134,13 @@ def generate_plot_mpl(
         else:
             self.ax = ax
 
+        # Get the labels in the color bar. This determines the number of unique colors
+        self._make_colorbar_labels()
         # Set colormap to be used in elements of emission and absorption plots
-        self.cmap = cm.get_cmap(cmapname, self.elements.size)
+        self.cmap = cm.get_cmap(cmapname, len(self._species_name))
+        self._show_colorbar_mpl()
 
```

Comment:
Add self.colorlist or something similar that tracks the species colors for the cmap.

Reply:
This is implemented now.